rust_snuba/src/processors/metrics_summaries.rs

```diff
@@ -53,7 +55,7 @@ struct InputMessage {
     #[serde(default)]
     duration_ms: u32,
     project_id: u64,
-    #[serde(default = "default_retention_days")]
```

Comment:
`#[serde(default)]` should do the right thing AFAIK, not sure why we hooked up `default_retention_days` here to begin with

Reply:
default_retention days was applying the 90 days previously

Reply:
your change makes `retention_days` a mandatory argument. you need `serde(default)` in order to default it to `None`, which afaik gets coerced into 90d later

Reply:
wdym? we definitely have `serde(default)` in this processor, on duration_ms. where are we missing it?

Reply:
in metrics_summaries.rs it is. so i guess the two consumers we have don't agree on which fields are optional. but that's imo besides the point for this PR

Reply:
There's examples outside of this topic though. In outcomes for example there are many optional fields: https://github.com/getsentry/snuba/blob/master/rust_snuba/src/processors/outcomes.rs#L77-L80

There is no issue parsing them when they are missing, such as in this test: https://github.com/getsentry/snuba/blob/master/rust_snuba/src/processors/outcomes.rs#L92-L98

Reply:
ugh, my bad, i think i misunderstood how this works. TIL